parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -71,7 +81,24 @@ public boolean equals(Object obj) {
 
   @Override
   public String toString() {
-    return "Binary{" + length() + " bytes, " + Arrays.toString(getBytes()) + "}";
+    return "Binary{" +
+        length() +
+        (isReused ? " reused": " constant") +
+        " bytes, " +
+        Arrays.toString(getBytesUnsafe())
+        + "}";
+  }
+
+  public Binary copy() {
+    if (isReused) {
+      return Binary.fromConstantByteArray(getBytes());
+    } else {
+      return this;
+    }
+  }
+
+  public boolean isReused() {
```

Comment:
can we add comments explaining what it means to be reused?
There's an implicit contract here right? This is for signaling who "owns" the backing byte array right?